src/scanspec/core.py

```diff
@@ -288,6 +292,29 @@ def if_instance_do(x: C, cls: type[C], func: Callable[[C], T]) -> T:
 AxesPoints = dict[Axis, npt.NDArray[np.float64]]
 
 
+class Slice(Generic[Axis]):
+    """Generalization of the Dimensions class.
+
+    Only holds information but no methods to handle it.
+    """
+
+    def __init__(
+        self,
+        axes: list[Axis],
+        midpoints: AxesPoints[Axis],
+        gap: GapArray,
+        lower: AxesPoints[Axis] | None = None,
+        upper: AxesPoints[Axis] | None = None,
+        duration: DurationArray | None = None,
```

Comment:
I don't think any of these can be None, as they are always expanded by `Dimension` and passed to this function

```suggestion
        lower: AxesPoints[Axis],
        upper: AxesPoints[Axis],
        duration: DurationArray,
```

Reply:
That makes sense! But I'm wondering, we're still able to class `Path.consume()` into a Dimension that doesn't have the `duration` defined, so I think we could keep the option for `duration` to be None by default.